src/trust-region/basic-trust-region.jl

```diff
@@ -0,0 +1,40 @@
+export TrustRegion
+
+"""Basic trust region type."""
+type TrustRegion <: AbstractTrustRegion
+  initial_radius :: Float64
+  radius :: Float64
+  max_radius :: Float64
+  acceptance_threshold :: Float64
+  increase_threshold :: Float64
+  decrease_factor :: Float64
+  increase_factor :: Float64
+  ratio :: Float64
```

Comment:
Should the ratio really be part of the structure? It's computed externally and isn't really a property of the trust region itself.